pkg/apis/kops/validation/aws.go

```diff
@@ -277,3 +282,24 @@ func awsValidateLoadBalancerSubnets(fieldPath *field.Path, spec kops.ClusterSpec
 
 	return allErrs
 }
+
+func awsValidateCPUCredits(fieldPath *field.Path, spec *kops.InstanceGroupSpec) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !featureflag.EnableLaunchTemplates.Enabled() {
+		allErrs = append(allErrs, field.Forbidden(fieldPath.Child("cpuCredits"), "launch templates must be enabled with `export KOPS_FEATURE_FLAGS=EnableLaunchTemplates`"))
+	}
+
+	allErrs = append(allErrs, IsValidValue(fieldPath.Child("cpuCredits"), spec.CPUCredits, []string{"standard", "unlimited"})...)
+	if spec.MixedInstancesPolicy != nil {
+		for i, instanceType := range spec.MixedInstancesPolicy.Instances {
```

Comment:
Also t4g

Reply:
Not sure if it applies to those instances or just to the main instance type, or how would be handled by non-burstable instances.

Reply:
ah you're right. In fact I can create a launch template using one non-burstable instance type and have credit specification set, and I can still create an ASG with no MIP. I guess we could drop the instance type checks here altogether.

Reply:
Good idea.